### DISCUSSION TITLE
[Question][Hw5] How difficult condition of filter can be?

### DISCUSSION THREAD

**Discussion by robotjoaa** [prosyslab-classroom/cs348-information-security#617] [Q&A]
I saw #143 and still I have some questions regarding complex conditions.
In the provided examples there were only simple ones like x > 0 and also in case of condition like x > y, we know variable x and y is directly related to the condition and it can be filtered. 

But for conditions like x + 1 > y, x+1 part will be substituted to different llvalue and put into the instruction for branch node of graph. I would like to ask whether the target behavior is filtering memory of x even though it is indirectly related to the variables of condition. 

If the condition gets more difficult like (x+y+z > 5), it should go to the before instructions to inversely calculate dependencies. 

Is this the case or do we only have to consider filtering variables that directly appear on the condition? (i.e. filter done only looking at the one llvalue and variables in its operand)

**Comment by RiceBiscuits** [comment]
You don't need to filter each x, y, and z.
